### PR TITLE
Apply LUA api providers before loading script.

### PR DESCRIPTION
**Pull request by ConnorBP**
Fixes issue where api is not available during script initialization. https://github.com/makspll/bevy_mod_scripting/issues/97

PR is not yet complete. There is still a `callback error` caused by: error converting Lua nil to userdata to investigate which happens when the api function is called.

### CODE REVIEW

**Comment by ConnorBP** [comment]
Found what is causing the NIL error. It happened upon fetching the bevy "world" from globals. It is not set on script load execution, only after event execution is fired. This makes it so that any API function which accesses world state cannot be used during script load. To fix this requires a refactor involving changing the `script_add_synchronizer` into an `exclusive system` with a `SystemState`

**Comment by ConnorBP** [comment]
Update: the other issues have their own concerns that should be in another PR. But the api registration issue is solved by this PR and should be considered Complete.

I will open a different PR for potential api refactoring for connecting callback functions with event signals.

File: languages/bevy_mod_scripting_lua/src/lib.rs [hunk 1]
```diff
@@ -113,17 +113,24 @@ impl<A: LuaArg> ScriptHost for LuaScriptHost<A> {
         #[cfg(not(feature = "unsafe_lua_modules"))]
         let lua = Lua::new();
 
-        lua.load(script)
+        // init lua api before loading script
+        let mut lua = Mutex::new(lua);
```
Comment (makspll) [inline languages/bevy_mod_scripting_lua/src/lib.rs:117]: Could we wrap in the mutex at the end instead?

**Review by makspll — CHANGES_REQUESTED** [review]
Looks good, just one small nit

**Comment by ConnorBP** [comment]
> Looks good, just one small nit

`providers.attach_all()' requires a mutex as it's argument.

**Comment by makspll** [comment]
Sounds good, let's just fix CI and happy to merge

**Review by makspll — APPROVED** [review]

**Pull request merged**